cpp/oneapi/dal/backend/primitives/sort/sort.hpp

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <typename Float>
+struct float2uint_map;
+
+template <>
+struct float2uint_map<float> {
+    using integer_t = std::uint32_t;
+};
+
+template <>
+struct float2uint_map<double> {
+    using integer_t = std::uint64_t;
+};
+
+/// @tparam Float Floating-point type used for storing input values
+/// @tparam Index Integer type used for storing input indices
+template <typename Float, typename Index = std::uint32_t>
+class radix_sort_indices_inplace {
+    using radix_integer_t = typename float2uint_map<Float>::integer_t;
+
+public:
+    /// Performs initialization of auxiliary variables and required auxiliary buffers
+    ///
+    /// @param[in]  queue The queue
+    /// @param[in]  elem_count  The number of elements in input vector
+    radix_sort_indices_inplace(sycl::queue& queue, std::int64_t elem_count = 0);
```

Comment:
If queue is stored by value in this object, why you require to pass it as non-const reference?

Reply:
fixed